freshbooks.js

```diff
@@ -10,84 +10,99 @@ function FreshBooks(access_token, refresh_token) {
 }
 
 FreshBooks.prototype.me = function(callback) {
-  this._request('auth/api/v1/users/me', null, function(err, data) {
+  this._get('auth/api/v1/users/me', function(err, data) {
     if (err) return callback(err);
 
     var me = data.response;
-    var required_props = ['id', 'first_name', 'last_name', 'email'];
-    required_props.forEach(function(prop) {
-      if (!(prop in me))
-        throw new Error('Required property "' + prop + '" not found in /me: ' + JSON.stringify(me));
-      
-      var expected_type = prop == 'id' ? 'number' : 'string';
-      if (typeof me[prop] != expected_type)
-        throw new Error('me.' + prop + ' is not a ' + expected_type + ': ' + JSON.stringify(me[prop]));
-    });
-
-    if (!me.business_memberships || !_.isArray(me.business_memberships))
-      throw new Error('Expected me.business_memberships to be an array: ' + JSON.stringify(me.business_memberships));
-
-    var business_ids = me.business_memberships.map(function(biz_membership) {
-      if (!biz_membership.business || !_.isObject(biz_membership.business))
-        throw new Error('Expected me.business_memberships to have a .business object: ' + JSON.stringify(biz_membership));
-
-      if (typeof biz_membership.business.id != 'number')
-        throw new Error('Expected me.business_memberships.business.id to be a number: ' + JSON.stringify(biz_membership.business));
-
-      return biz_membership.business.id;
-    });
-
-    var me = _.pick(me, required_props);
-    me.business_ids = business_ids;
```

Comment:
@schuttsm: My intent (with `pick()` and `business_ids`) was to make the API flatter (easier to use) and to restrict our use of the API to the subset of properties/objects that are being tested. We should probably also allow access to underlying methods (perhaps underscore-prefixed?) that provide "raw" access to the API, but a lot of the value we're adding is providing a documented & tested stable subset of an alpha API.

Going a little further, I think we should add `active_business_ids`, which provides a little more value, filtering the business IDs to the ones with active subscriptions (since querying a business with a lapsed subscription results in an error). I'm not sure if we also include `business_ids` or not -- perhaps we should, as it may make things a little less mysterious to an end user.

IOW, the more FreshBooks-specific logic, business rules, etc, that we can codify into this library (and out of our codebase) the better. It will make our app's codebase simpler and will provide value to others, so they don't have to learn the same things we've had to learn.

Reply:
@prust - that makes sense.

I was under the impression that this project was *only* intended to tests around an api to let us know when it changed.

If you want to massage the api in a way that makes more logical sense, that's fine - I think I'd favor keeping the raw data available & add other methods for that.